### DISCUSSION TITLE
Ninetendo wifu

### DISCUSSION THREAD

**Discussion by GeekyG4mer** [rafaelvcaetano/melonDS-android#317] [Q&A]
Is nintendo wifi supported, or will it be supported in the future. I am asking because the emulator can find melonAP as an access point. But can't connect. Do I have to tweak some settings?

**Comment by aimdloop** [comment]
> Is nintendo wifi supported, or will it be supported in the future. I am asking because the emulator can find melonAP as an access point. But can't connect. Do I have to tweak some settings?

Nope nintendo wifi is not yet implemented.More information #38

**Comment by GeekyG4mer** [comment]
Hey, I was interested in the Nintendo wifi because I wanted to access the underground in pokemon platinum. When I tried it was possible I have found fossils and stuff and only issue I faced was when exiting the underground the game crashed only once with 5-7 successful exits. This is the only android ds emulator to do that . In my opinion this crushes all other competition you have in the android platform.

**Reply by rafaelvcaetano** [reply]
Wi-Fi is not out of the question. But it's not easy. I've been looking into it since I've started developing this port but I still don't have a good solution for it and I will keep the investigation ongoing.

**Reply by GeekyG4mer** [reply]
What do you mean by solution? It is flawless already.

**Reply by rafaelvcaetano** [reply]
Online multiplayer is not working yet, just as you said in the original post. So that's what still needs to be worked on.

**Reply by aimdloop** [reply]
> Hey, I was interested in the Nintendo wifi because I wanted to access the underground in pokemon platinum. 

@GeekyG4mer Well is it require ninetendo wfc?? 
Update- It don't require ninetendo wfc connection (online multiplayer).
It requires ds wireless communication (local multiplayer).
Check this out-https://youtu.be/Qlc8GGxoteI

**Reply by aimdloop** [reply]
@rafaelvcaetano I am not sure. Can't u do this without patching the rom? I don't know what wfcpatcher does with the rom. Can't u take a idea from it and add it in melonds?

**Reply by rafaelvcaetano** [reply]
Sure, it can be done. I can do what WFCPatcher is doing as the ROM is loaded. I just need to basically copy its code. I know the project's license allows it but I'll try to reach the author for permission.

**Reply by aimdloop** [reply]
Thats awesome 😘😍

**Reply by ghost** [reply]
I dont think thats the solution but keep going.
I know for flycast the dreamcast emulator they use a library/stack called "picotcp".
The purpose is for emulating the dreamcast modem and it works on Android as well pretty well.
The code is here if  you are interested: https://github.com/flyinghead/flycast/tree/master/core/network
There is also Citra MMJ builds for android that also got working online multiplayer but I think the source code is not shared for that not sure but it works on Android as well.
I hope this info could help @rafaelvcaetano .

**Reply by ghost** [reply]
Also found out that Dosbox-x uses libslirp to implement ethernet networking.
Might be useful to check out as well:
https://github.com/joncampbell123/dosbox-x/tree/master/src/misc

**Reply by Zephyrxx0** [reply]
> There is also Citra MMJ builds for android that also got working online multiplayer but I think the source code is not shared for that not sure but it works on Android as well.

No citra mmj does not support online multiplayer, it supports local mulitplayer

**Reply by ghost** [reply]
> > There is also Citra MMJ builds for android that also got working online multiplayer but I think the source code is not shared for that not sure but it works on Android as well.
> 
> No citra mmj does not support online multiplayer, it supports local mulitplayer

Yeah but I mean you can play it online lol (works over internet).

**Comment by ashgx** [comment]
I tested it in version 1.3.0. I use WfcPatcher1.6 in Pokemon platinum ROM. After that, just like using the computer simulator, set dns178.62.43.212, and then the connection is successful. I can download mystery gifts successfully, but I can't use GTS and exchange.. I have a question: originally, WFC is not required for computer simulator and ROM on R4 card, but why does Android version of melonds need WFC? Is it because of the previous version setting?
WfcPatcher1.6：https://github.com/AdmiralCurtiss/WfcPatcher/releases/tag/v1.6

**Reply by GeekyG4mer** [reply]
So the game needs to be patched?

**Reply by ashgx** [reply]
> 那么游戏需要打补丁吗？

yes，If the game is not patched, it will not be connected to the network.(but using the melonds on the computer to play the game, the game doesn't need to be patched at all)

**Reply by aimdloop** [reply]
> yes，If the game is not patched, it will not be connected to the network.(but using the melonds on the computer to play the game, the game doesn't need to be patched at all)

Really!!Can u connect to the MelonAP and get mystery gift on melonds android?
I wanted to patch the rom using wfcpatcher bt never do that. Coz i didn't realize it will work. Can u check this dns workes and send a screenshot? 
Primary dns-164.132. 44.106 
Secondary dns-1.1.1.1 or 8.8.8.8

**Comment by ashgx** [comment]
preparatory stage：
![1](https://user-images.githubusercontent.com/81661156/117558797-e1915f00-b0b2-11eb-8f57-4dc4b8e3ef14.png)
Just drag and drop dumped NDS games onto WFC patcher1.6 to patch them，e.g. generate  Pokemon - Black Version (USA) (NoSSL).nds，and modify game archive name
![2](https://user-images.githubusercontent.com/81661156/117558804-ee15b780-b0b2-11eb-8c6f-f2d85cc0b758.png)
Import game files and Archives into mobile phone
![3](https://user-images.githubusercontent.com/81661156/117558805-f110a800-b0b2-11eb-959a-db6d7d3c4446.png)
The demonstration is as follows：https://youtu.be/5ysKGH5kspo

**Reply by aimdloop** [reply]
Wow man gg🤩. I never thought this will work. I thought online multiplayer is not yet implemented.It needs to be patched right. I guess only mystery gift works for now. Global trade or battle gives communication error. Thanks for checking that dns.

**Reply by ghost** [reply]
Wait what? it works? now i wonder if the retroarch core also could do this...

**Comment by ashgx** [comment]
In the previous test, I took my dad's Android phone.  I usually use  iphone, and my dad needs to use his phone, so I can't continue the test.
Recently, I entrusted my friend to test GTS. He tested GTS and found that Gen4 can test GTS, but Gen5 can't. He doesn't know why.
![1](https://user-images.githubusercontent.com/81661156/117786202-d89dba80-b277-11eb-887c-085826d3bd91.jpg)
![2](https://user-images.githubusercontent.com/81661156/117786219-db98ab00-b277-11eb-8296-91482a352026.jpg)

**Reply by aimdloop** [reply]
Wow😲😮.Then global trade is possible for gen 4 games. Idk what's wrong in gen 5.

**Reply by ashgx** [reply]
> Wow😲😮.Then global trade is possible for gen 4 games. Idk what's wrong in gen 5. I wish i could test.
> 
> Can u give me any patched rom link which you have? I know it's illegal bt i can't use my pc rn coz my exam😭

Pokemon - Black Version (USA, Europe) (NoSSL).zip
[removed]
Pokemon - SoulSilver Version (USA) (NoSSL).zip
[removed]

**Reply by rafaelvcaetano** [reply]
Guys, please do not share piracy here. I will remove the links from the the previous comment.

**Reply by ashgx** [reply]
> Guys, please do not share piracy here. I will remove the links from the the previous comment.

All right .I didn't know before. I won't share it later

**Reply by ghost** [reply]
So I guess everything thats related to matchmaking is not working...
The issue is there.

**Reply by aimdloop** [reply]
>Pokemon - Black Version (USA, Europe) (NoSSL).zip
[removed]
Pokemon - SoulSilver Version (USA) (NoSSL).zip
[removed]

Sorry i didn't notice.

>Guys, please do not share piracy here. I will remove the links from the the previous comment.

Ok sorry.😔

**Reply by aimdloop** [reply]
I tested that ninetendo wfc connection is being established.
I can get mystery gift, do some online battle and trade with my friends on gen 4 games.Here is a video getting mystery gift on pokemon soul silver-https://youtu.be/GWWvlEjrN24
Unfortunately i can't make a screen record of global trade. When i turn on screen recorder it gives communication error.
I didn't test on gen 5 games. And thanks @ashgx for sharing this workaround.😊😊

**Reply by Zephyrxx0** [reply]
> > Wow😲😮.Then global trade is possible for gen 4 games. Idk what's wrong in gen 5. I wish i could test.
> > Can u give me any patched rom link which you have? I know it's illegal bt i can't use my pc rn coz my exam😭
> 
> Pokemon - Black Version (USA, Europe) (NoSSL).zip
> [removed]
> Pokemon - SoulSilver Version (USA) (NoSSL).zip
> [removed]

Hey can you give me too?? I dont own a pc so i can get one. Can send somewhere else

**Reply by RayyanAnsari** [reply]
> > > Wow😲😮.Then global trade is possible for gen 4 games. Idk what's wrong in gen 5. I wish i could test.
> > > Can u give me any patched rom link which you have? I know it's illegal bt i can't use my pc rn coz my exam😭
> > 
> > Pokemon - Black Version (USA, Europe) (NoSSL).zip
> > [removed]
> > Pokemon - SoulSilver Version (USA) (NoSSL).zip
> > [removed]
> 
> Hey can you give me too?? I dont own a pc so i can get one. Can send somewhere else

Please don't share ROMS and other copyrighted materials.

What you could try is compiling that patcher yourselves inside an application known as 'Termux'.

**Comment by aimdloop** [comment]
People are saying that they can connect on other games like mario kart and play with others too. Maybe not all games. It can be also use to trade with myself instead of local multiplayer through GTS. It gives more stable connection than local multiplayer.

**Reply by ghost** [reply]
I have someone try mario kart 7 and it didnt work so not sure about that lol.

**Reply by ghost** [reply]
Found a sure way to make it work (i think).
Use WfcPatcher --domain option from command line and use "--domain wiimmfi.de" on PC to make the rom be compatible with wiimmfi without using any DNS.
This at least makes MKDS to work and I could be in the lobby and search for other people.

**Reply by aimdloop** [reply]
Yes mario kart is not working properly. It's connecting but can't matchmaking-https://youtu.be/MItpp8yTN_Y

**Comment by aimdloop** [comment]
⏪ Replacement of Ninetendo WFC⏩
             ✌️Primary DNS✌️
Wiimmfi/Riiconnect24-
164.132.44.106 (Recommend)
95.217.77.181
46.4.79.141
95.217.77.151
Poke Classic Network/Kaeru WFC-178.62.43.212(Recommend)
Twilit WFC- 34.66.49.81
BenFi- 24.218.177.103
        ✌️Secondary DNS✌️
 8.8.8.8 / 1.1.1.1 / 0.0.0.0 / same as primary / keep it blank

⏪Gen 4 Pokemon Games(Tested on 164.132.44.106 DNS server)⏩

✌️Working✌️
1- Mystery Gift- https://youtu.be/GWWvlEjrN24
2- GTS-https://youtu.be/5AmDw_HLtG0
We can also use this method to trade with myself. It's more stable than local multiplayer.
3- Wifi battle tower- https://youtu.be/t162Zygbxk0
It's full of cheaters.

😔Not working/Somewhat working😔
1- Wifi battle and trade with friends-https://youtu.be/vvrtwqr3Y2U
I invited my friend for a battle. We were in a different network. Then we register our friends code in pal pad. Then we enter the wifi club room. We tried several times to battle and trade. But it seems we can see each other bt can't battle with each other. Hope we will be able to wifi battle and trade in the future.
2- Wifi plaza- https://youtu.be/sUk-xM4blpo
https://youtu.be/01CudzwUvqY

⏪Gen 5 Pokemon Games(Tested on 164.132.44.106 DNS Server⏩

✌️Working✌️
1. Mystery Gift

😔Not working/Somewhat working😔
1- GTS- https://youtu.be/kJ8iVotjC48
https://youtu.be/qWSa57oYbA4
2- Random wifi battles- https://youtu.be/H00FLnK10nE
Get error code- 13282

🤟Important Links🤟
Polaris Servers List-https://github.com/barronwaffles/dwc_network_server_emulator/wiki/List-of-Servers

Wiimmfi-
https://wiimmfi.de/
https://wiimmfi.de/error
https://wiimmfi.de/stat?m=28

GTS Listing(Gen 4 & 5 PKMN)-
http://pkmnclassic.net/gts/

AltWFC
Error Codes/Troubleshooting: https://github.com/polaris-/dwc_network_server_emulator/wiki/Troubleshooting

https://forum.wii-homebrew.com/index.php/Thread/51738-Collecting-Error-Codes/#post590262

**Reply by GeekyG4mer** [reply]
I think this information should have a discussion of its own, instead of being buried deep in this mistyped discussion. So that more people can test and provide feedback.

**Reply by aimdloop** [reply]
> I think this information should have a discussion of its own, instead of being buried deep in this mistyped discussion. So that more people can test and provide feedback.

It's pointless rn because many things won't work using this workaround. We can do that when a better solution will come.

**Reply by ashgx** [reply]
> > I think this information should have a discussion of its own, instead of being buried deep in this mistyped discussion. So that more people can test and provide feedback.
> 
> It's pointless rn because many things won't work using this workaround. We can do that when a better solution will come.

Is there a disord？About melonds Android?

**Reply by chocoblin** [reply]
can we use this in drastic?

**Reply by ashgx** [reply]
> can we use this in drastic?

I tried the above method and it didn't work.（I tested it with drastic the year before.）

**Comment by ghost** [comment]
The Switch port gonna have libslirp integrated as well now.
Talked to the dev and they said it will be easier to get it to work on other ports now as well (as long as they got a socket interface).
https://github.com/RSDuck/melonDS

**Comment by Fewnity** [comment]
Any news about the Wifi support?

**Reply by Swappo91** [reply]
yes rafael... you have news? the dream is play online poke woth phones bro. I know you can do that you are a monster broo

**Comment by ASGAMING009** [comment]
Btw If anyone's wanna Try Wfc patcher and patch their Rom but dont have Pc to do Soo...
I have a method So you can Do The Wfc Connection Without patching the rom

![Screenshot_2023-02-08-13-25-57-332_me magnum melonds](https://user-images.githubusercontent.com/68546736/217468828-da38cb94-715c-47bf-ba74-0f3ec1fe538f.png)

You can use this cheat so you dont have to patch stuff and it will work same as patching....
Sidenote!
No wifi isn't compatible With the emulator just yet
You can only do Gts And Some Other stuff
Wifi plaza Will give you an error
Wifi club will work but ya cant do nothing no battle nor trade
The creator Of This Port is Trying his Hardest To Implement Wifi So We should Respect That And Dont Pressure Him!
Hope you all understand.
Thx!

**Reply by Swappo91** [reply]
i think they are actually dont try any stuff to get it work.

**Reply by ASGAMING009** [reply]
Wdym

**Reply by Swappo91** [reply]
Actually the developer are not working to solve the wfc problem. 
I think solve this issue is essential to grow the usage of this emu.

All people want to play mariokart ds for example on phone in multiplayer. Or exchange pokemons for example.

I think soon are coming other emulators that solve that issue and this emu just become old.

I think solve that problem are not too hard but i dont understand why rafael dont try to tell some help for example.

**Reply by ASGAMING009** [reply]
Its not that Easy To do So... We Just have to Wait...

**Reply by Swappo91** [reply]
Any news?

**Reply by aimdloop** [reply]
@Swappo91 Bro why don't u just make it for yourself. We are lucky that someone even thinking about it in year 2023. If it's impossible to implement then we have to accept it. Don't bother the developer or discourage them.

**Reply by ASGAMING009** [reply]
He's A 9 Y/o Who doesn't Know anything about coding and im pretty sure he cant Even Open a zip file...
